pkg/documents/documents.go

```diff
@@ -101,43 +104,53 @@ func uploadCustomerDocument(logger log.Logger, repo DocumentRepository, bucketFa
 			return
 		}
 
-		file, _, err := r.FormFile("file")
-		if file == nil || err != nil {
+		// if r.Body is larger than maxFormSize an error will be returned when the body is read
+		r.Body = http.MaxBytesReader(w, r.Body, maxFormSize)
+		file, fileHeader, err := r.FormFile("file")
+		if err != nil {
+			if strings.Contains(err.Error(), "request body too large") {
+				logger.Log("documents", "max form size exceeded", "customerID", customerID, "requestID", requestID)
+				moovhttp.Problem(w, fmt.Errorf("request body exceeds maximum size of %dMB", maxFormSize/(1<<20)))
+				return
+			}
+			logger.Log("documents", "error reading form file", "error", err, "customerID", customerID, "requestID", requestID)
 			moovhttp.Problem(w, fmt.Errorf("expected multipart upload with key of 'file' error=%v", err))
 			return
 		}
 		defer file.Close()
 
-		// Detect the content type by reading the first 512 bytes of r.Body (read into file as we expect a multipart request)
-		buf := make([]byte, 512)
-		if _, err := file.Read(buf); err != nil && err != io.EOF {
-			logger.Log("documents", fmt.Sprintf("failed to peek: %v", err), "customerID", customerID)
-			moovhttp.Problem(w, err)
+		if fileHeader.Size > maxDocumentSize {
+			logger.Log("documents", "max file size exceeded", "customerID", customerID, "requestID", requestID)
+			moovhttp.Problem(w, fmt.Errorf("file exceeds maximum size of %dMB", maxDocumentSize/(1<<20)))
```

Comment:
Almost skimmed over that byte shift. Can we do this math in a helper? 

or when we consolidate file uploads offer this as a helper

Reply:
I could create a type for it, like `sizeLimit`, then implement methods like `String()`...

Reply:
Sounds great

Reply:
I added it to this file/package for now, but I agree it should be in some kind of file upload helper package in the future.